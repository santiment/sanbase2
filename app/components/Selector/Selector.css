.selector {
  display: flex;
  user-select: none;
  cursor: pointer;
  min-width: 32px;
  text-align: center;
  border-radius: 4px;
  padding: 10px 25px;
}

.selector-item {
  cursor: pointer;
  min-width: 32px;
  text-align: center;
  border-radius: 4px;
  padding: 0 5px;
}

.selector .selector-item--selected {
  background: rgba(38, 43, 51, 0.8);
  color: white !important;
}

.selector .selector-item--disabled {
  color: #888;
  cursor: not-allowed;
}

.night-mode .selector,
.night-mode .selector-item--selected {
  color: #a0a0a0;
  background-color: #252932;
}

.night-mode .selector-item--disabled {
  color: #595959;
}
