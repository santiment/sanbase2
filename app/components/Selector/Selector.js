import React, { Component } from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './Selector.css'

export const SelectorItem = ({
  isSelected = false,
  value = 'all',
  name = 'all',
  setFilter,
  disabled = false
}) => (
  <div
    className={cx({
      'selector-item': true,
      'selector-item--selected': isSelected,
      'selector-item--disabled': disabled
    })}
    onClick={() => !disabled && setFilter(value)}
  >
    {name}
  </div>
)

export class Selector extends Component {
  state = {
    selected: this.props.defaultSelected
  }

  static defaultProps = {
    defaultSelected: undefined,
    options: [],
    disabled: false
  }

  static propTypes = {
    defaultSelected: PropTypes.string,
    options: PropTypes.array,
    onSelectOption: PropTypes.func,
    disabled: PropTypes.bool
  }

  static getDerivedStateFromProps (nextProps, prevState) {
    if (nextProps.defaultSelected !== prevState.selected) {
      return {
        selected: nextProps.defaultSelected
      }
    }
    return prevState
  }

  onSelectOption = newOption => {
    this.setState({ selected: newOption }, () => {
      this.props.onSelectOption(newOption)
    })
  }

  render () {
    const { selected } = this.state
    const { options, disabled } = this.props
    const nameOptions = this.props.nameOptions || options
    return (
      <div className='selector'>
        {options.map((option, index) => (
          <SelectorItem
            key={option}
            name={nameOptions[index]}
            isSelected={selected === option}
            value={option}
            setFilter={this.onSelectOption}
            disabled={disabled}
          />
        ))}
      </div>
    )
  }
}

export default Selector
