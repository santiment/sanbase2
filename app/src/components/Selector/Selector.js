import React, { Component } from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './Selector.css'

export const SelectorItem = ({
  isSelected = false,
  value = 'all',
  setFilter,
  disabled = false
}) => (
  <div
    className={cx({
      'selector-item': true,
      'selector-item--selected': isSelected,
      'selector-item--disabled': disabled
    })}
    onClick={() => !disabled && setFilter(value)}
  >
    {value}
  </div>
)

class Selector extends Component {
  state = {
    selected: this.props.defaultSelected
  }

  static defaultProps = {
    defaultSelected: undefined,
    options: [],
    disabled: false
  }

  static propTypes = {
    defaultSelected: PropTypes.string,
    options: PropTypes.array,
    onSelectOption: PropTypes.func,
    disabled: PropTypes.bool
  }

  onSelectOption = newOption => {
    this.setState({ selected: newOption }, () => {
      this.props.onSelectOption(newOption)
    })
  }

  render () {
    const { selected } = this.state
    const { options, disabled } = this.props
    return (
      <div className='selector'>
        {options.map(option => (
          <SelectorItem
            key={option}
            isSelected={selected === option}
            value={option}
            setFilter={this.onSelectOption}
            disabled={disabled}
          />
        ))}
      </div>
    )
  }
}

export default Selector
