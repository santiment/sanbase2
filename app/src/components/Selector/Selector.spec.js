/* eslint-env jest */
import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import Selector, { SelectorItem } from './Selector'

const defaultOptions = ['1w', '1m', '3m', '6m', '1y', 'all']

describe('Selector component', () => {
  it('smoke', () => {
    const mockCb = jest.fn()
    const wrapper = shallow(
      <Selector options={defaultOptions} onSelectOption={mockCb} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should be called onSelectOption, if Selector option is clicked', () => {
    const mockCb = jest.fn()
    const wrapper = mount(
      <Selector options={defaultOptions} onSelectOption={mockCb} />
    )
    const firstSelectorItem = wrapper.find('.selector-item').at(2)
    firstSelectorItem.simulate('click')
    expect(mockCb).toHaveBeenCalled()
  })

  it('should change selected option, after click on not selected item', () => {
    const mockCb = jest.fn()
    const wrapper = mount(
      <Selector options={defaultOptions} onSelectOption={mockCb} />
    )
    expect(
      wrapper
        .find(SelectorItem)
        .at(2)
        .props().isSelected
    ).toBeFalsy()
    expect(
      wrapper
        .find('.selector-item')
        .at(2)
        .text()
    ).toBe('3m')
    const firstSelectorItem = wrapper.find('.selector-item').at(2)
    firstSelectorItem.simulate('click')
    expect(wrapper.state('selected')).toBe('3m')
  })

  it('should have selected option is 3m, if defaultSelected is 3m', () => {
    const wrapper = shallow(
      <Selector options={defaultOptions} defaultSelected='3m' />
    )
    expect(
      wrapper
        .find(SelectorItem)
        .at(2)
        .props().isSelected
    ).toBeTruthy()
  })
})
