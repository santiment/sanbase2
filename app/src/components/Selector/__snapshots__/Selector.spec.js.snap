// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector component smoke 1`] = `
<div
  className="selector"
>
  <Component
    disabled={false}
    isSelected={false}
    key="1w"
    name="1w"
    setFilter={[Function]}
    value="1w"
  />
  <Component
    disabled={false}
    isSelected={false}
    key="1m"
    name="1m"
    setFilter={[Function]}
    value="1m"
  />
  <Component
    disabled={false}
    isSelected={false}
    key="3m"
    name="3m"
    setFilter={[Function]}
    value="3m"
  />
  <Component
    disabled={false}
    isSelected={false}
    key="6m"
    name="6m"
    setFilter={[Function]}
    value="6m"
  />
  <Component
    disabled={false}
    isSelected={false}
    key="1y"
    name="1y"
    setFilter={[Function]}
    value="1y"
  />
  <Component
    disabled={false}
    isSelected={false}
    key="all"
    name="all"
    setFilter={[Function]}
    value="all"
  />
</div>
`;
